commonknowledge/wagtail/embed_providers/shiny.py

```diff
@@ -0,0 +1,56 @@
+
+import re
+from html.parser import HTMLParser
+from urllib.parse import urlparse
+from wagtail.embeds.finders.base import EmbedFinder
+import requests
+from django.utils.text import slugify
+
+
+class ShinyFinder(EmbedFinder):
+    SHINY_URL_PATTERN = re.compile(
+        r'https://stop-watch.shinyapps.io/(.+)/?')
```

Comment:
This regex could be simplified to

```regex
https://stop-watch.shinyapps.io/(.+)
```

This is because `.+` will include any trailing slash (`/`), so `/?` (optional `/`) is unecessary.